### PR TITLE
Setup script for Smaract 3DOF

### PR DESCRIPTION
**Pull request by amuskens**

### CODE REVIEW

File: src/odemis/driver/test/smaract_test.py [hunk 1]
```diff
@@ -446,7 +446,6 @@ def test_reference_cancel(self):
             # procedure is started, so this doesn't work.
             # TODO: extend the simulator to report axis is referenced only at the end of referencing.
             f = self.dev.reference()
-            time.sleep(0.1)
```
Comment (pieleric) [inline src/odemis/driver/test/smaract_test.py:449]: Leave this line here. It has nothing to do with the script (and the line is actually needed).

File: scripts/smaract_reference_mode.py [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+                'range': [-3e-3, 3e-3],
+                'unit': 'm',
+                'channel': 1,
+            },
+            'z': {
+                'range': [-3e-3, 3e-3],
+                'unit': 'm',
+                'channel': 2,
+            },
+        },
```
[the remark this answers is not in the document]
Comment (amuskens) [reply scripts/smaract_reference_mode.py:57]: On the fly?

[the remark this answers is not in the document]
Comment (pieleric) [reply scripts/smaract_reference_mode.py]: As in:
```python
axes = {
    str(c): {"range": [-1, 1], "unit": "m", "channel": c}
    for c in channels
}
```

File: util/smaractconfig.py [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+import logging
+import time
+
+NUM_CHANNELS = 3
+
+CONFIG_3DOF = {"name": "3DOF",
+        "role": "stage",
+        "ref_on_init": False,
+        # Use the locator for real hardware when necessary. 
+        "locator": "network:sn:MCS2-00001604",
```
Comment (pieleric) [inline util/smaractconfig.py:37]: Drop it from here.

File: util/smaractconfig.py [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+Odemis. If not, see http://www.gnu.org/licenses/.
+'''
+
+from odemis.driver import smaract
+import sys
+import argparse
+import logging
+import time
+
+NUM_CHANNELS = 3
```
Comment (pieleric) [inline util/smaractconfig.py:31]: Don't hard code this value. Just compute the axes based on the axes in arguments.

File: util/smaractconfig.py [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+    # Set up logging before everything else
+    if options.loglev < 0:
+        logging.error("Log-level must be positive.")
+        return 127
+    loglev_names = (logging.WARNING, logging.INFO, logging.DEBUG)
+    loglev = loglev_names[min(len(loglev_names) - 1, options.loglev)]
+    logging.getLogger().setLevel(loglev)
+
+    if options.locator:
+        CONFIG_3DOF['locator'] = options.locator
```
Comment (pieleric) [inline util/smaractconfig.py:145]: This option is required, so no need for a `if`.

File: util/smaractconfig.py [hunk 6]
```diff
@@ -0,0 +1,160 @@
...
+    logging.info("New position: %s", dev.position.value)
+    logging.info("Complete.")
+
+
+def main(args):
+
+    parser = argparse.ArgumentParser(prog="smaractconfig",
+                                     description='Run a one time config for the SmarAct ')
+
+    parser.add_argument('--locator', required=True, type=str, help='Run the operations with the simulator', action="store")
```
Comment (pieleric) [inline util/smaractconfig.py:120]: That help string is not really clear. Indicate how it should look like "network:sn:MCS2-00001601" and that "fake" can be used for a simulator.

File: util/smaractconfig.py [hunk 7]
```diff
@@ -0,0 +1,160 @@
...
+        logging.error("Log-level must be positive.")
+        return 127
+    loglev_names = (logging.WARNING, logging.INFO, logging.DEBUG)
+    loglev = loglev_names[min(len(loglev_names) - 1, options.loglev)]
+    logging.getLogger().setLevel(loglev)
+
+    if options.locator:
+        CONFIG_3DOF['locator'] = options.locator
+
+    dev = smaract.MCS2(**CONFIG_3DOF)
```
Comment (pieleric) [inline util/smaractconfig.py:147]: You can write `smaract.MCS2(locator=locator, axes=axes, **CONFIG_3DOF)`
And you should compute axes just here, based on which axes are specified in the options.

File: util/smaractconfig.py [hunk 8]
```diff
@@ -0,0 +1,160 @@
...
+    """
+    Set calibration option bits to calibration_options. See section 2.7.1 in programmer manual
+    0 is for a default calibration
+    The following flags can be used:
+    smaract.SA_CTLDLL.SA_CTL_CALIB_OPT_BIT_DIRECTION
+    smaract.SA_CTLDLL.SA_CTL_CALIB_OPT_BIT_DIST_CODE_INV_DETECT
+    smaract.SA_CTLDLL.SA_CTL_CALIB_OPT_BIT_ASC_CALIBRATION
+    smaract.SA_CTLDLL.SA_CTL_CALIB_OPT_BIT_REF_MARK_TEST
+    smaract.SA_CTLDLL.SA_CTL_CALIB_OPT_BIT_LIMITED_TRAVEL_RANGE
+    """
```
Comment (pieleric) [inline util/smaractconfig.py:59]: Make it a comment, not a long string.

**Comment by amuskens** [comment]
Should be ready to merge now.

**Pull request merged**